numba/tests/test_looplifting.py

```diff
@@ -201,6 +205,9 @@ def test_lift4(self):
     def test_lift5(self):
         self.check_lift_ok(lift5, (types.intp,), (123,))
 
+    def test_lift_issue2561(self):
+        self.check_lift_ok(lift_issue2561, (types.List,), ([],))
```

Comment:
`types.List` here will trigger an assertion in the type-inference because it is not an instance of `types.Type`.  That assertion is why the function is dropping into object-mode and into looplifting, but it seems like a bug in the test.